packages/compass-aggregations/src/components/stage-wizard/index.tsx

```diff
@@ -176,13 +179,20 @@ export default connect(
       .find((x): x is StoreStage => x.type === 'stage' && !x.disabled);
 
     const mappedInitialFields = initialFields.map(
-      (x: { name: string }) => x.name
+      (x: { name: string; description: string }) => ({
+        name: x.name,
+        // parsed schema has the bson type Object replaced with
+        // Document to avoid collision with JS Objects but that
+        // shouldn't be a problem for us because we use these
+        // as string values alongside well defined casters.
+        type: x.description === 'Document' ? 'Object' : x.description,
+      })
     );
     const previousStageFields = getSchema(previousStage?.previewDocs ?? []);
 
     const fields =
       previousStageFields.length > 0 && autoPreview
-        ? previousStageFields
+        ? previousStageFields.map((name) => ({ name, type: 'String' }))
```

Comment:
I looked into this and it turned out that the documents that we are providing to getSchema are not parsed schema documents from compass-field-store but, either the preview documents from previous stage or an empty list which is correct because we want to show fields from previous stage.

I looked into refactoring the part of `compass-field-store` that does this schema parsing and it seems doable but I will need to spend some more time on it, if this is how we should proceed ahead.

Reply:
as discussed in conversation, @himanshusinghs will do a follow up to add bson type in `get-schema` util.